numba/cloudpickle/cloudpickle.py

```diff
@@ -220,6 +220,7 @@ def _lookup_module_and_qualname(obj, name=None):
         # types.ModuleType. The other possibility is that module was removed
         # from sys.modules after obj was created/imported. But this case is not
         # supported, as the standard pickle does not support it either.
+        del obj.__module__
```

Comment:
```suggestion
        obj.__module__ = None
```

Some object type forbids the deletion of `__module__`. For example, any class type will reject this.

On the other hand, it is odd for this function to mutate the object.